### DISCUSSION TITLE
output-location support for Beamer

### DISCUSSION THREAD

**Discussion by jlperla** [quarto-dev/quarto-cli#6926] [Feature Requests]
### Description

In #6918  it appears there are no clean workarounds.

The requested feature is simple:  beamer support for the `output-location:slide` feature which works well in `revealjs`.  For example, the following works great:

    ---
    title: "Test Doc"
    format: revealjs 
    execute: 
      cache: true    
    echo: true
    ---
    
    ## Some Code
    
    Here is some python code
    
    ```{python}
    #| output-location: slide
    import matplotlib.pyplot as plt
    
    plt.plot([1, 2, 3, 4])
    plt.ylabel('some numbers')
    plt.show()
    ```

where I want to swap to `format: beamer` (and also make sure jupyter notebook output isn't broken).

**Comment by cderv** [comment]
Thanks for opening the feature request ! 

For reference, related to past discussion that could be revisited
* https://github.com/quarto-dev/quarto-cli/issues/664

**Comment by jlperla** [comment]
@cderv   Oh yeah, the output to the side was my next question!  I think that slowly sneaking in revealjs features for beamer output would be really nice.  FOr many fields PDF is the standard, and for some good reasons.

**Reply by cscheid** [reply]
> I think that slowly sneaking in revealjs features for beamer output would be really nice

FWIW, we think the size of our `quarto` + `beamer` user base is under 5% of `quarto` + `revealjs`. Unfortunately, we have limited development time and other things are higher in our priority. With that said, if you want to contribute such improvements in a PR, we'd be happy to take them.

**Comment by jlperla** [comment]
@cscheid  I played around a little with `revealjs` and it is impressive, especially for code display.  The tradeoff seems to be that while beamer had limited code display support it showed all of the math correctly (not surprisingly since it generates latex), but I am having a lot of issues with revealjs and display of math across operating systems and browsers.  Issues do not occur in Jupyter.  I will post a more specific issue but curious if general issues with math display in revealjs which has a bigger workaround.

**Reply by cderv** [reply]
> I am having a lot of issues with revealjs and display of math across operating systems and browsers. Issues do not occur in Jupyter. I will post a more specific issue but curious if general issues with math display in revealjs which has a bigger workaround.

Please do open issues and feedback about this. RevealJS math is processed by Mathajx, but KaTeX is also an option. There is some updates to do (https://github.com/quarto-dev/quarto-cli/issues/4493) but maybe this is different. 

THanks

**Comment by jlperla** [comment]
@cderv   Will do.  At this point adding `https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js` as you pointed out seems to solve my problems.  It seems like Quarto 1.4 is a monster release, but hopefully it won't be too long as things like updated mathjax would have a big impact.
